### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Core

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
There are 3 things left TODO:

- [ ] in `Core.id_hom` the types don't match in ` Iso.hom (𝟙 X) = 𝟙 X `, I suspect one of the `𝟙` gets interpreted the wrong way?
- [ ] The mathlib3port gave 
```
instance : Faithful (inclusion C) where
``` 
and in mathlib3 it just worked without a proof, here it doesn't. I started unfolding the statement a but, didn't finish
 - [ ] Two problems in the proof of  `ofEquivFunctor`

**Comment by mattrobball** [comment]
> There are 3 things left TODO:
> 
> * [ ]  in `Core.id_hom` the types don't match in `Iso.hom (𝟙 X) = 𝟙 X`, I suspect one of the `𝟙` gets interpreted the wrong way?
> * [ ]  The mathlib3port gave
> 
> ```
> instance : Faithful (inclusion C) where
> ```
> 
> and in mathlib3 it just worked without a proof, here it doesn't. I started unfolding the statement a but, didn't finish
> 
> * [ ]  Two problems in the proof of  `ofEquivFunctor`

If you said `(X : Core C)` Lean had a hard time getting back to `(X:C)`. For the instance, `Iso.ext` was almost exactly what is wanted. I suspect this was seen somehow.

File: Mathlib/CategoryTheory/Core.lean [hunk 1]
```diff
@@ -17,46 +17,46 @@ import Mathlib.CategoryTheory.Types
...
 but this is not functorial with respect to `F`.
 -/
 
-
 namespace CategoryTheory
 
 universe v₁ v₂ u₁ u₂
 
--- morphism levels before object levels. See note [category_theory universes].
+-- morphism levels before object levels. See note [CategoryTheory universes].
```
Comment (urkud) [inline Mathlib/CategoryTheory/Core.lean:32]: The note is called `category_theory universes`. If you want to rename it, then please open a separate PR.

**Review by urkud — COMMENTED** [review]
I have 2 minor comments, otherwise LGTM. Thanks!
bors d+

**Pull request closed without merging**